DB set up

TorusTredent opened the pull request.

anle751 committed:
Create entities

TorusTredent committed:
add dependencies and properties for liquibase

TorusTredent committed:
add data to changeSet files

TorusTredent committed:
add table creation structures to sql files

TorusTredent committed:
add table-relationships files

anle751 committed:
Creating entity relationships according to the database structure

TorusTredent committed:
+ moved sql files from
+ add relations between tables

anle751 committed:
fix mapping User to Task

TorusTredent committed:
+ change xml to yaml
+ rename, relocate files

TorusTredent committed:
* add file drop_relationships
* refact files

TorusTredent committed:
add liquibase properties for work liquibase commands

TorusTredent committed:
Merge branch 'database' into liquibase

TorusTredent committed:
comment password
fix naming and user fields

TorusTredent committed:
fix applicatein.properties

VolgaDaniluk commented on application/build.gradle:
```diff
@@ -2,20 +2,45 @@ plugins {
+    liquibaseRuntime 'org.liquibase:liquibase-groovy-dsl:3.0.2'
+    liquibaseRuntime 'info.picocli:picocli:4.6.3'
+    liquibaseRuntime group: 'javax.xml.bind', name: 'jaxb-api', version: '2.3.1'
+    implementation 'mysql:mysql-connector-java:8.0.30'
```
why do u need to implement it the second time?

VolgaDaniluk commented on application/build.gradle:
```diff
@@ -2,20 +2,45 @@ plugins {
+            url "jdbc:mysql://127.0.0.1:3306/kids_app"
+            driver "com.mysql.cj.jdbc.Driver"
+            username "root"
+            password "12344321"
```
what is this credentionals for? Is it our DB or yours?

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Photo.java:
```diff
@@ -0,0 +1,13 @@
+@Setter
+@Data
+public class Photo {
+    private String path;
```
where are id?

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Request.java:
```diff
@@ -0,0 +1,47 @@
+        result = 31 * result + (reward != null ? reward.hashCode() : 0);
+        result = 31 * result + (user != null ? user.hashCode() : 0);
+        return result;
+    }
```
can we use lombok insted of this? (both equals and hashcode)

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Request.java:
```diff
@@ -0,0 +1,47 @@
+@Entity
+public class Request implements Serializable {
+
+    @Enumerated(EnumType.STRING)
```
use enam

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Request.java:
```diff
@@ -0,0 +1,47 @@
+    @Id
+    @OneToOne(optional = false)
+    @JoinColumn(nullable = false)
+    private Reward reward;
```
add empty line after

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Request.java:
```diff
@@ -0,0 +1,47 @@
+@Setter
+@AllArgsConstructor
+@Entity
+public class Request implements Serializable {
```
Serializable? Explain

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Reward.java:
```diff
@@ -0,0 +1,42 @@
+    private Long points;
+    private String familyId;
+
+    @Enumerated(EnumType.STRING)
```
Enam

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Reward.java:
```diff
@@ -0,0 +1,42 @@
+    @Column(name = "reward_status")
+    private REWARD_STATUS rewardStatus;
+
+    @Enumerated(EnumType.STRING)
```
Enam

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Request.java:
```diff
@@ -0,0 +1,47 @@
+    private REQUEST_STATUS requestStatus;
+
+    @Id
+    @OneToOne(optional = false)
```
In what side we have to add "mapped by"?

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Reward.java:
```diff
@@ -0,0 +1,42 @@
+    @Column(name = "reward_type")
+    private REWARD_TYPE rewardType;
+
+    @OneToOne(orphanRemoval = true,mappedBy = "reward")
```
In what side we have to add "mapped by"?

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Reward.java:
```diff
@@ -0,0 +1,42 @@
+    @AttributeOverrides({
+            @AttributeOverride(name = "path", column = @Column(name = "path"))
+    })
+    private List<Photo> photo;
```
compare this implementation add using only JoinTable

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Task.java:
```diff
@@ -0,0 +1,48 @@
+    private String title;
+    private String description;
+    private Long points;
+    private Date deadline;
```
are you sure about the type?

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Task.java:
```diff
@@ -0,0 +1,48 @@
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "task_status")
+    private TASK_STATUS taskStatus;
```
Enaum

add empty line after

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Task.java:
```diff
@@ -0,0 +1,48 @@
+    @Enumerated(EnumType.STRING)
+    @Column(name = "task_status")
+    private TASK_STATUS taskStatus;
+    private boolean daily;
```
rename (add "is")

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Task.java:
```diff
@@ -0,0 +1,48 @@
+    private TASK_STATUS taskStatus;
+    private boolean daily;
+
+    @Enumerated(EnumType.STRING)
```
enum

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Task.java:
```diff
@@ -0,0 +1,48 @@
+    private TASK_TYPE taskType;
+
+    @Column(name = "family_id")
+    private String familyId;
```
where is the relation ?

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Task.java:
```diff
@@ -0,0 +1,48 @@
+    @AttributeOverrides({
+            @AttributeOverride(name = "path", column = @Column(name = "path"))
+    })
+    private List<Photo> photo;
```
compare this implementation add using only JoinTable

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/User.java:
```diff
@@ -0,0 +1,51 @@
+    private Long id;
+    private String email;
+
+    @Enumerated(value = EnumType.STRING)
```
enum

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/User.java:
```diff
@@ -0,0 +1,51 @@
+    @Enumerated(value = EnumType.STRING)
+    private USER_ROLE user_role;
+
+    @Enumerated(value = EnumType.STRING)
```
enum

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/User.java:
```diff
@@ -0,0 +1,51 @@
+    private USER_STATUS user_status;
+    private Long points;
+    private String name;
+    private String familyId;
```
where is the relation?

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/User.java:
```diff
@@ -0,0 +1,51 @@
+    private String familyId;
+
+//    @OneToOne(mappedBy = "user")
+//    private Password pass;
```
what is it?)

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/enums/REQUEST_STATUS.java:
```diff
@@ -0,0 +1,6 @@
+
+public enum REQUEST_STATUS {
+    APPROVED,
+    REJECT
```
add status "pending"

VolgaDaniluk commented on application/src/main/java/com/jungeeks/entitiy/Reward.java:
```diff
@@ -0,0 +1,42 @@
+
+    @Enumerated(EnumType.STRING)
+    @Column(name = "reward_type")
+    private REWARD_TYPE rewardType;
```
we don't need it

VolgaDaniluk commented on application/src/main/resources/application.properties:
```diff
@@ -1,2 +1,17 @@
+spring.datasource.username=root
+spring.datasource.password=12345678Root
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.MySQL8Dialect
+spring.jpa.properties.hibernate.show_sql=true
```
what is this cred for?

VolgaDaniluk commented on application/src/main/resources/db/changelog/v-1-0-0/sql/V20220902_drop_relationships.sql:
```diff
@@ -0,0 +1,27 @@
+    drop foreign key family_task_sec_users_family_id_fk;
+
+alter table kids_app.family_task
+    drop foreign key family_task_task_id_fk;
```
what is this file for? Why you need it?

VolgaDaniluk commented on application/src/main/resources/db/changelog/v-1-0-0/sql/V20220902_create_relationships.sql:
```diff
@@ -0,0 +1,37 @@
+alter table kids_app.request
+    add constraint request_sec_users_id_fk
+        foreign key (user_id) references kids_app.sec_users (id);
+
```
don't use shema name

TorusTredent committed:
+ add Family.java
+ change hibernate relationships between tables

TorusTredent committed:
models updated

TorusTredent committed:
+ add new category fields
+ add CATEGORY_TYPE

anle751 committed:
* Creating Repositories
* Add field 'id' to Request entity
* Add annotationProcessor 'org.projectlombok:lombok:1.18.24' to build.gradle

anle751 committed:
* Add ConfirmationToken

TorusTredent committed:
+ add SocialCredentials (google)
+ add FamilyTask to make a Task for templates
+ moved repository to com/java/jungeeks
+ fix entities

TorusTredent committed:
+ refact liquibase for new entities
+ fix entities

anle751 committed:
* add in FamilyTask to field 'task' 'CascadeType.persist'
* add repositories
* remove empty lines
